pkg/scheduler/preemption/preemption.go

```diff
@@ -277,7 +321,11 @@ func workloadFits(wlReq cache.ResourceQuantities, cq *cache.ClusterQueue) bool {
 				// Workload doesn't request this flavor.
 				continue
 			}
-			if cqFlvsUsage[flvLimits.Name]+flvReq > flvLimits.Min {
+			limit := flvLimits.Min
+			if flvLimits.Max != nil && allowBorrowing {
+				limit = *flvLimits.Max
```

Comment:
you also need to consider the cohort's requestable and usage

Reply:
This is checked already below.